static/src/javascripts/projects/commercial/modules/messenger/passback.ts

```diff
@@ -44,22 +83,92 @@ const init = (register: RegisterListener): void => {
 					.pubads()
 					.getSlots()
 					.find((s) => s.getSlotElementId() === slotIdWithPrefix);
+
 				if (slot) {
 					/**
-					 * All viewports >= 0x0 use fixed size 300x250
+					 * Copy the targeting from the previous slot
+					 */
+					const pageTargeting = getValuesForKeys(
+						window.googletag.pubads().getTargetingKeys(),
+						(key) => window.googletag.pubads().getTargeting(key),
+					);
+					const slotTargeting = getValuesForKeys(
+						slot.getTargetingKeys(),
+						(key) => slot.getTargeting(key),
+					);
+					const allTargeting: Array<[string, string[]]> = [
+						...pageTargeting,
+						...slotTargeting,
+						['passback', [source]],
+						['slot', ['inline']],
+					];
+
+					/**
+					 * Create a new ad slot element
 					 */
-					slot.defineSizeMapping([[[0, 0], [[300, 250]]]]);
+					const passbackElement = document.createElement('div');
+					passbackElement.id = `${slotIdWithPrefix}--passback`;
+					passbackElement.classList.add('ad-slot', 'js-ad-slot');
+					passbackElement.setAttribute('aria-hidden', 'true');
+					// position absolute to position over the container slot
+					passbackElement.style.position = 'absolute';
+					// account for the ad label
+					passbackElement.style.top = `${labelHeight}px`;
+					// take the full width so it will center horizontally
+					passbackElement.style.width = '100%';
+					slotElement.insertAdjacentElement(
+						'beforeend',
+						passbackElement,
+					);
+
+					/**
+					 * Define and display a new slot
+					 */
+					window.googletag.cmd.push(() => {
+						const passbackSlot = googletag.defineSlot(
+							slot.getAdUnitPath(),
+							[300, 250],
+							passbackElement.id,
+						);
+						passbackSlot?.addService(window.googletag.pubads());
+						allTargeting.forEach(([key, value]) => {
+							slot.setTargeting(key, value);
+						});
+						googletag.display(passbackElement.id);
```

Comment:
Since this logic lives outside of our existing ad-rendering flow, am I right in thinking we don't get the typical features such as refreshing, Prebid, etc? We didn't support this with previous passbacks, so it's obviously not an issue, but is this something we anticipate wanting to support in the future? And would it be simple/difficult using this approach?

Reply:
> see if we can make the existing ad stack more reusable then?

Good shout, if this ends up being a motivation to break up the flow into clearer stages that'd be great. It'd be nice if here we could opt into certain features and ignore others, say _"setup Prebid on this slot, and make it refresh every 30 seconds, but I don't care about any of the googletag or targeting setup because I'm handling that myself"_

Reply:
>  break up the flow into clearer stages that'd be great.

This PR has made me think more about what that might look like and I'm in total agreement! We've talked before about what a greenfield ad stack might look like.

Despite my initial rejection of classes I am coming around to something you've suggested before i.e. having an Advert class that is configurable and encapsulates everything about an ad. I think that's worth trying to get a feel for the ergonomics. The current stack is hard to reason about as it's a spiders web of granular functions invoked without any determinable lifecycle.

It would great if we could encapsulate the ad lifecycle in a declarative way.

I guess it's not too far away from a component model of the world that we've spiked before: https://github.com/guardian/commercial-playground/tree/main/react-ads but with regular JS rather than React components.

Reply:
> Despite my initial rejection of classes I am coming around to something you've suggested before i.e. having an Advert class that is configurable and encapsulates everything about an ad. I think that's worth trying to get a feel for the ergonomics. The current stack is hard to reason about as it's a spiders web of granular functions invoked without any determinable lifecycle.

This is interesting, I think it's definitely worth exploring. Even if not as a `class` but some clear way that you can hook into different steps of the lifecycle.